data_structures/disjoint_set/disjoint_set.py

```diff
@@ -0,0 +1,67 @@
+class Node:
+    def __init__(self, data):
+        self.data = data
+
+
+def make_set(x):
+    """
+    make x as a set.
+    """
+    # rank is the distance from x to its' parent
+    # root's rank is 0
+    x.rank = 0
+    # p is x's parent
+    x.p = x

```

Comment:
Let's call this __x.parent__ instead of __x.p__ then the comment above can be removed.  Single character variable names are old school.